File: src/main/java/cn/techfan/quickdoc/entities/FsDirectory.java
package cn.techfan.quickdoc.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@AllArgsConstructor
@ToString
@Data
public class FsDirectory {
    @Id
    private final Long id;
    private String path;
    private Long parentId;
    private FsOwner[] owners = null;
}

File: src/main/kotlin/com/neofinance/quickdoc/common/entities/FsDirectory.kt
package com.neofinance.quickdoc.common.entities

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class FsDirectory(@Id val id: Long,
                       var path: String,
                       var parentId: Long,
                       var owners: Array<FsOwner>? = null)
